openvm/src/lib.rs

```diff
@@ -742,7 +729,52 @@ impl tracing::field::Visit for PgoData {
 // A Layer that collects data we are interested in using for the pgo from the trace fields.
 #[derive(Clone)]
 struct PgoCollector {
-    pc: Arc<Mutex<Vec<usize>>>,
+    step: usize,
+    pc_base: usize,
+    map: Arc<Mutex<Vec<u32>>>,
+}
+
+impl PgoCollector {
+    fn new<F>(program: &Program<F>) -> Self {
+        let max_pc = program.instructions_and_debug_infos.len() * program.step as usize;
+        // create a map with max_pc entries initialized to 0
+        let map = Arc::new(Mutex::new(vec![0; max_pc]));
+        Self {
+            map,
+            step: program.step as usize,
+            pc_base: program.pc_base as usize,
+        }
+    }
+
+    fn take(self) -> HashMap<u32, u32> {
+        // take the map from the mutex
+        let map = std::mem::take(&mut *self.map.lock().unwrap());
+
+        // Turn the map into a HashMap of (pc_index, count)
+        map.chunks(self.step)
+            .enumerate()
+            .filter_map(|(pc_index, chunk)| {
+                // Sanity check that all counts are zero except the first one
```

Comment:
Not sure I get this here. All counts of what are 0 except first what?

Reply:
Will do this, it's cleaner.